logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/GetMessageByIdUseCaseTest.kt

```diff
@@ -0,0 +1,96 @@
+package com.wire.kalium.logic.feature.message
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.framework.TestConversation
+import com.wire.kalium.logic.framework.TestMessage
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.test_util.TestKaliumDispatcher
+import com.wire.kalium.util.KaliumDispatcher
+import io.mockative.Mock
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+
+class GetMessageByIdUseCaseTest {
+
+    private val testDispatchers: KaliumDispatcher = TestKaliumDispatcher
+
+    @Test
+    fun givenMessageAndConversationId_whenInvokingUseCase_thenShouldCallMessageRepository() = runTest(testDispatchers.io) {
+        val messageId = "messID"
+        val conversationId = TestConversation.ID
+
+        val (arrangement, getMessageByIdUseCase) = Arrangement()
+            .withRepositoryMessageByIdReturning(conversationId, messageId, Either.Left(CoreFailure.Unknown(null)))
+            .arrange()
+
+        getMessageByIdUseCase(conversationId, messageId)
+
+        verify(arrangement.messageRepository)
```

Comment:
Idk which solution is better but can this check be moved to the other two tests?

Reply:
I try avoiding mixing different assertions as much as I can.

Having fewer per test, trying to grouped them based on the behaviour we are usually has the benefit that when it fails we can read the test name and easily understand what is failing.

Otherwise we need to dive deeper to udnerstand which one of the assertions is failing and why it is being tested.

Imagine we don't have `givenMessageAndConversationId_whenInvokingUseCase_thenShouldCallMessageRepository`, and we move this assertion to `givenRepositoryFails_whenInvokingUseCase_thenShouldPropagateTheFailure` and then this verification fails.

This means that "should propagate failure" is failing.
But it's failing because we're not passing the proper parameter to the repository?! The test doesn't reflect the assertion.